src/sequence/mod.rs

```diff
@@ -719,22 +723,162 @@ mod tests {
     }
 
     // verify data convergence on a set of replicas and with the expected length
-    fn verify_data_convergence(replicas: &[&Sequence], expected_len: u64) {
+    fn verify_data_convergence(replicas: Vec<Sequence>, expected_len: u64) {
         // verify replicas have the expected length
         // also verify replicas failed to get with index beyond reported length
         let index_beyond = SequenceIndex::FromStart(expected_len);
-        for r in replicas {
+        for r in &replicas {
             assert_eq!(r.len(), expected_len);
             assert_eq!(r.get(index_beyond), None);
         }
 
         // now verify that the items are the same in all replicas
         for i in 0..expected_len {
             let index = SequenceIndex::FromStart(i);
-            let r0_entry = replicas[0].get(index);
-            for r in replicas {
-                assert_eq!(r0_entry, r.get(index));
+            let r0_entry = &replicas[0].get(index);
+            for r in &replicas {
+                assert_eq!(r0_entry, &r.get(index));
             }
         }
     }
+
+    /// Generate a vec of Sequence replicas of some length
+    fn generate_replicas(quantity: i32, xorname: XorName, tag: u64) -> Result<Vec<Sequence>> {
```

Comment:
Not sure it makes sense to generate replicas all with same xorname and tag, I think we can either generate those values here randomly, or have these function to receive `Vec<(XorName, u64)>` as arg. instead, and it generates one replica per vec item.

Reply:
But isn't it a replica of a given piece of data in the network? Like, each replica will be managing data of xorname `X` and tag: `64`, that's how we know it's the same data... Otherwise what's to converge upon, it could be a completely different piece of data, no?

Reply:
yeah, you are right, i had to refresh my memory with this :)